### PR TITLE
feat(018): hotstring CLI create + list commands

### PR DESCRIPTION
**Pull request by s205109**
## Summary
- Adds `ahkflow hotstring new` and `ahkflow hotstring list` (with `--json`).
- Replaces `NullAuthTokenProvider` with `EnvVarAuthTokenProvider` (reads `AHKFLOW_TOKEN`).
- Implements `IProfilesApiClient` (deferred from 017); registers typed HttpClient pipeline for hotstrings + profiles.

## Test plan
- [x] `dotnet test` green (49 unit + 15 integration in CLI tests; full repo suite passing).
- [x] `ahkflow --help`, `ahkflow hotstring --help`, `ahkflow hotstring new --help` render correctly.
- [ ] Manual: against a running TEST API with a valid token, create + list hotstrings, both human + `--json`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Tools/AHKFlowApp.CLI/Output/HotstringTableFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        foreach (Guid id in dto.ProfileIds)
+        {
+            if (names.TryGetValue(id, out string? name)) resolved.Add(name);
+            else unresolved++;
+        }
+
+        if (resolved.Count == 0) return $"{dto.ProfileIds.Length} profiles";
+
+        string head = string.Join(", ", resolved.Take(3));
+        int more = resolved.Count - 3 + unresolved;
```
[the remark this answers is not in the document]
Comment (s205109) [reply src/Tools/AHKFlowApp.CLI/Output/HotstringTableFormatter.cs]: Fixed.  — the previous formula underflowed to 0 when resolved < 3 with unresolved > 0.

File: tests/AHKFlowApp.CLI.Tests/Services/EnvVarAuthTokenProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public async Task GetTokenAsync_EnvVarUnsetOrBlank_ThrowsNotAuthenticated(string? value)
+    {
+        using var _ = new EnvVarScope(Var, value);
+        EnvVarAuthTokenProvider sut = new();
+
+        Func<Task> act = () => sut.GetTokenAsync(CancellationToken.None);
+
+        (await act.Should().ThrowAsync<NotAuthenticatedException>())
+            .WithMessage("Not signed in. Set AHKFLOW_TOKEN environment variable to a bearer token.");
+    }
```
[the remark this answers is not in the document]
Comment (s205109) [reply tests/AHKFlowApp.CLI.Tests/Services/EnvVarAuthTokenProviderTests.cs:36]: Not an issue here. xUnit runs tests within the same class sequentially — no two tests from EnvVarAuthTokenProviderTests can run concurrently. EnvVarScope already handles save/restore. The only cross-class risk would be if another test class mutated AHKFLOW_TOKEN in parallel, but none does.

File: docs/superpowers/specs/2026-05-09-018-hotstrings-cli-support-design.md [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+| `--profile` | `-p` | string | no | — | Filter by profile name (single value, resolved to Guid). If omitted, shows all. |
+| `--search` | `-s` | string | no | — | LIKE-match on trigger or replacement. |
+| `--page` | — | int | no | 1 | Page number (1-indexed). |
+| `--page-size` | — | int | no | 50 | Items per page (1–200). |
+| `--json` | — | bool | no | false | Emit response as JSON instead of human table. |
+
+**Behavior:**
+- GETs `/api/v1/hotstrings?profileId=<guid>&search=<text>&page=<n>&pageSize=<n>`.
+- On success (200): prints human table (default) or JSON array.
+- Profile names in table column are resolved via a preliminary `GET /api/v1/profiles` call (single round-trip). If a profile ID has no name in the response, falls back to `<id>` literal.
```
[the remark this answers is not in the document]
Comment (s205109) [reply docs/superpowers/specs/2026-05-09-018-hotstrings-cli-support-design.md]: Fixed in spec: the partial-resolve case shows resolved names with a +N more suffix; the all-unresolved case shows '<n> profiles'. The '<id> literal' fallback never existed in the implementation.

File: docs/superpowers/specs/2026-05-09-018-hotstrings-cli-support-design.md [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+|------|-------|------|----------|---------|-------|
+| `--profile` | `-p` | string | no | — | Filter by profile name (single value, resolved to Guid). If omitted, shows all. |
+| `--search` | `-s` | string | no | — | LIKE-match on trigger or replacement. |
+| `--page` | — | int | no | 1 | Page number (1-indexed). |
+| `--page-size` | — | int | no | 50 | Items per page (1–200). |
+| `--json` | — | bool | no | false | Emit response as JSON instead of human table. |
+
+**Behavior:**
+- GETs `/api/v1/hotstrings?profileId=<guid>&search=<text>&page=<n>&pageSize=<n>`.
+- On success (200): prints human table (default) or JSON array.
```
[the remark this answers is not in the document]
Comment (s205109) [reply docs/superpowers/specs/2026-05-09-018-hotstrings-cli-support-design.md]: Fixed in spec: updated to 'JSON object (PagedList envelope with items, page, pageSize, totalCount)'.
